### PR TITLE
adds CoffStore for persistence #22

### PR DESCRIPTION
**Pull request by brennanMKE**
Defines protocol `CoffStore` with an implementation named `FilesystemCoffStore` which uses `Codable` for serializing data to JSON under Documents/CoffStore. There is an extension stubbed out for Combine. For some reason the build works and the app runs but tests are failing to build for me.

### CODE REVIEW

File: Shared/CoffAppApp.swift [hunk 1]
```diff
@@ -14,7 +13,6 @@ struct CoffAppApp: App {
     var body: some Scene {
         WindowGroup {
             ContentView()
-                .environment(\.managedObjectContext, persistenceController.container.viewContext)
```
Comment (rodericj) [inline Shared/CoffAppApp.swift:17]: I think I'm missing how the environment changed here. Was this just not used anywhere? Why are we able to remove this line?

**Review by mcritz — CHANGES_REQUESTED** [review]
I love the idea of a CoffStore protocol. I especially see the benefits for a Combined-based approach.

Buuuuut… high level, the point of this project is to try out modern tech for building an app. What I’d love is to see all the callback based APIs refactored into `async` or `async throws`. I’m 100% on board with raising the build target to iOS 15 and learning new patterns. Like Mel Brooks says, “It’s good to be the king!”

Or, per our in person discussion, we could have done this in Combine which is really good for this type of data flow. (Fetching local results, publishing, receiving network results, removing duplicates, publishing diffs, etc.) I see the stub there, but since it doesn’t do anything, just remove it. I suppose we could still do async with Combine. I’ve never tried it, and I’d be down to see if that’s a worthwhile pattern.

Hope you don’t mind a little refactoring. But I think it’ll be worth it to get that real world async / await experience.

File: Shared/CoffAppApp.swift (hunk 1 above)
Comment (mcritz) [reply Shared/CoffAppApp.swift:17]: @rodericj It wasn’t used anywhere. Just template cruft.

**Comment by mcritz** [comment]
Merging in now. We can refactor from here.

**Pull request merged**